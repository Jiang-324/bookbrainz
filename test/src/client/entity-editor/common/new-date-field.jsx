// eslint-disable-next-line import/no-unassigned-import
import '../../../../../enzyme.config';
import * as React from 'react';
import {expect, use} from 'chai';
import DateField from '../../../../../src/client/entity-editor/common/new-date-field';
import {jestSnapshotPlugin} from 'mocha-chai-jest-snapshot';
import {mount} from 'enzyme';
import {spy} from 'sinon';


use(jestSnapshotPlugin());

describe('DateField', () => {
	let wrapper;
	it('should render correctly', () => {
		wrapper = mount(<DateField/>);
		expect(wrapper.html()).toMatchSnapshot();
	});
	it('should display given date correcly', () => {
		const isoDate = '+000012-12-14';
		const dummyLabel = 'SomeLabel';
		wrapper = mount(<DateField defaultValue={isoDate} empty={false} error={false} label={dummyLabel}/>);
		expect(wrapper.find('label span').text()).to.equal(dummyLabel);
		expect(wrapper.find('FormControl.year-field').props().value).to.equal('0012');
		const obj = wrapper.find('FormControl.other-date-field');
		expect(obj.at(0).props().value).to.equal('12');
		expect(obj.at(1).props().value).to.equal('14');
	});
	it('should call onChange callback with correct arguments', () => {
		const onChangeSpy = spy();
		wrapper = mount(<DateField onChangeDate={onChangeSpy}/>);
		const yearInput = wrapper.find('.year-field input');
		yearInput.simulate('change', {target: {value: '2022'}});
		expect(onChangeSpy.calledOnce).to.be.true;
		expect(onChangeSpy.args[0][0]).to.equal('+002022');
		const otherInput = wrapper.find('.other-date-field input');
		otherInput.at(0).simulate('change', {target: {value: '12'}});
		expect(onChangeSpy.calledTwice).to.be.true;
		expect(onChangeSpy.args[1][0]).to.equal('+002022-12');
		otherInput.at(1).simulate('change', {target: {value: '14'}});
		expect(onChangeSpy.calledThrice).to.be.true;
		expect(onChangeSpy.args[2][0]).to.equal('+002022-12-14');
	});
});
