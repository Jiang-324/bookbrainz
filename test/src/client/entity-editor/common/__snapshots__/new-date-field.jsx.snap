// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField should render correctly 1`] = `"<div><div class=\\"d-none form-group\\"><label class=\\"form-label\\"><span class=\\"text-warning\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"exclamation-triangle\\" class=\\"svg-inline--fa fa-exclamation-triangle fa-w-18 margin-left-0-5\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\"><path fill=\\"currentColor\\" d=\\"M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z\\"></path></svg><span class=\\"text-warning\\"> Are you sure? You entered a date in the future! </span></span></label><div class=\\"responsive-date-field input-group\\"><input maxlength=\\"4\\" placeholder=\\"YYYY\\" style=\\"width: 4.5em;\\" type=\\"text\\" class=\\"year-field form-control\\" value=\\"\\"><div class=\\"input-group-append\\" style=\\"margin-right: -1px;\\"><span class=\\"input-group-text\\" style=\\"padding: 0px 0.5em;\\">-</span></div><input maxlength=\\"2\\" placeholder=\\"MM\\" type=\\"text\\" class=\\"other-date-field form-control\\" value=\\"\\"><div class=\\"input-group-append\\" style=\\"margin-right: -1px;\\"><span class=\\"input-group-text\\" style=\\"padding: 0px 0.5em;\\">-</span></div><input maxlength=\\"2\\" placeholder=\\"DD\\" type=\\"text\\" class=\\"other-date-field form-control\\" value=\\"\\"><div class=\\"input-group-append\\"><div class=\\"react-datepicker-wrapper\\"><div class=\\"react-datepicker__input-container\\"><button style=\\"line-height: 1.75; padding: 0.375em 0.938em;\\" value=\\"\\" type=\\"button\\" class=\\"btn btn-info\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"calendar-alt\\" class=\\"svg-inline--fa fa-calendar-alt fa-w-14 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z\\"></path></svg></button></div></div></div></div></div></div>"`;
